stock_safe_scrap/models/stock_scrap.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2024 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class StockScrap(models.Model):
+
+    _inherit = "stock.scrap"
+
+    in_progress_picking_ids = fields.Many2many(
+        comodel_name="stock.picking",
+        compute="_compute_in_progress_picking_ids",
+        help="This indicates that an operation is in progress for the product and "
+        "the source location.",
+    )
+
+    @api.model
+    def _scrap_origin_common_keys(self):
+        """Return the list of fields that are used to select scrap from move lines."""
+        return [
+            "location_id",
+            "product_id",
+            "lot_id",
+            "package_id",
+            "owner_id",
+            "company_id",
+        ]
+
+    @api.depends("product_id", "location_id")
+    def _compute_in_progress_picking_ids(self):
+        """
+        Compute the pickings that are in progress
+        for the origin locations of the scraps.
+        """
+        move_lines = self.env["stock.move.line"].search(
+            [
+                ("state", "not in", ("done", "cancel")),
+                ("product_id", "in", self.product_id.ids),
+                ("location_id", "in", self.mapped("location_id").ids),
+                ("qty_done", ">", 0.0),
+            ]
+        )
+        for _key, scrap in self.partition(
+            lambda a_scrap: dict(
+                {key: a_scrap[key] for key in self._scrap_origin_common_keys()}
+            )
+        ).items():
+            scrap.in_progress_picking_ids = move_lines.filtered(
+                lambda line: all(
+                    (line[move_key] == scrap.mapped(move_key))
+                    for move_key in self._scrap_origin_common_keys()
+                )
+            ).picking_id
```

Comment:
@rousseldenis Can you explain this code? It's difficult to understand it. A comment would be welcome.